src/components/popover-container/popover-container.spec.tsx

```diff
@@ -785,6 +808,119 @@ describe("open state when click event triggered", () => {
     expect(onCloseFn).not.toHaveBeenCalled();
   });
 
+  it("should close the container and call onClose when controlled and escape key is pressed", () => {
+    const onCloseFn = jest.fn();
+    const MockWrapper = () => {
+      const [open, setOpen] = React.useState(true);
+
+      return (
+        <PopoverContainer
+          title="PopoverContainerSettings"
+          open={open}
+          onClose={(e) => {
+            setOpen(false);
+            onCloseFn(e);
+          }}
+        />
+      );
+    };
+    mount(<MockWrapper />);
+
+    document.dispatchEvent(
+      new KeyboardEvent("keydown", {
+        key: "Escape",
+        bubbles: true,
+      })
+    );
+    expect(onCloseFn).toHaveBeenCalled();
```

Comment:
**question**: even the test description, is this test missing an assertion for verifying the `PopoverContainer` has been closed?

```suggestion
    expect(wrapper.update().find(PopoverContainer).exists()).toBe(false);
    expect(onCloseFn).toHaveBeenCalled();
```

Reply:
Yeah I can put `expect(wrapper.update().find(PopoverContainerContent).exists()).toBe(false);` here as the `PopoverContainer` is in the DOM even when the container is closed.